packages/contractkit/src/wallets/signers/azure-key-vault-client.ts

```diff
@@ -0,0 +1,181 @@
+// @ts-ignore-next-line
+import { account as Account, nat as Nat } from 'eth-lib'
+import { DefaultAzureCredential } from '@azure/identity'
+import { KeyClient, CryptographyClient, KeyVaultKey } from '@azure/keyvault-keys'
+// @ts-ignore-next-line
+import BN, { BN } from 'bn.js'
+import { ec as EC } from 'elliptic'
+import { ecdsaRecover } from 'secp256k1'
+
+/**
+ * Provides an abstraction on Azure Key Vault for performing signing operations
+ */
+export class AzureKeyVaultClient {
+  private readonly vaultName: string
+  // Unique URI of the Azure Key Vault
+  private readonly vaultUri: string
+  private readonly credential: DefaultAzureCredential
+  private readonly keyClient: KeyClient
+  private readonly SIGNING_ALGORITHM: string = 'ECDSA256'
+
+  // 0x04 prefix indicates that the key is not compressed
+  // https://tools.ietf.org/html/rfc5480#section-2.2
+  private readonly publicKeyPrefix: number = 0x04
+
+  constructor(vaultName: string) {
+    this.vaultName = vaultName
+    this.vaultUri = `https://${this.vaultName}.vault.azure.net`
```

Comment:
Is this uri accessible regardless of being executed within an Azure network?

Reply:
Correct, this is the public endpoint for the KeyVault. It's authenticated via MSI in this case.